third-party/realdds/include/realdds/dds-stream.h

```diff
@@ -25,12 +32,24 @@ class dds_stream : public dds_stream_base
 protected:
     dds_stream( std::string const & stream_name, std::string const & sensor_name );
 
-    // dds_stream_base
 public:
-    bool is_open() const override;
-    bool is_streaming() const override;
+    bool is_open() const override { return !! _reader; }
+    virtual void open( std::string const & topic_name, std::shared_ptr< dds_subscriber > const & );
+    virtual void close();
+
+    bool is_streaming() const override { return _is_streaming; }
+    typedef std::function< void( topics::device::image && f) > on_data_available_callback;
+    void start_streaming( on_data_available_callback cb );
+    void stop_streaming();
 
     std::shared_ptr< dds_topic > const & get_topic() const override;
+
+protected:
+    void handle_frames();
+
+    std::shared_ptr< dds_topic_reader > _reader;
+    on_data_available_callback _on_data_available;
+    bool _is_streaming = false;

```

Comment:
This is pretty much the same as `_on_data_available` - not needed?

Reply:
Deleted. Using `_on_data_available` for streaming indication